junit-jupiter-engine/src/main/java/org/junit/jupiter/engine/execution/DefaultParameterResolver.java

```diff
@@ -0,0 +1,173 @@
+/*
+ * Copyright 2015-2018 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v2.0 which
+ * accompanies this distribution and is available at
+ *
+ * http://www.eclipse.org/legal/epl-v20.html
+ */
+
+package org.junit.jupiter.engine.execution;
+
+import static java.util.stream.Collectors.joining;
+import static java.util.stream.Collectors.toList;
+import static org.junit.platform.commons.util.ReflectionUtils.isAssignableTo;
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Executable;
+import java.lang.reflect.Parameter;
+import java.util.List;
+import java.util.Optional;
+
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.junit.jupiter.api.extension.ParameterContext;
+import org.junit.jupiter.api.extension.ParameterResolutionException;
+import org.junit.jupiter.api.extension.ParameterResolver;
+import org.junit.jupiter.engine.extension.ExtensionRegistry;
+import org.junit.platform.commons.logging.Logger;
+import org.junit.platform.commons.logging.LoggerFactory;
+import org.junit.platform.commons.util.Preconditions;
+
+public class DefaultParameterResolver {
```

Comment:
Since this not an implementation of `ParameterResolver`, we cannot call it `DefaultParameterResolver`.

Instead, we should name it something like `ParametersResolver`. The "s" is admittedly easy to overlook. So maybe we'll come up with something better later.

Reply:
Not sure if you're agreeing to extract this as a class now?  

Anyway, I've pushed some more changes with what I think you're saying.